junit-platform-console/src/main/java/org/junit/platform/console/options/AvailableOptions.java

```diff
@@ -10,273 +10,332 @@
+import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 
-import joptsimple.OptionParser;
-import joptsimple.OptionSet;
-import joptsimple.OptionSpec;
-import joptsimple.util.KeyValuePair;
-import joptsimple.util.PathConverter;
-
 import org.junit.platform.engine.discovery.ClassNameFilter;
 
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParameterException;
+import picocli.CommandLine.ParseResult;
+import picocli.CommandLine.Spec;
+
 /**
  * @since 1.0
  */
+@Command(name = "ConsoleLauncher", sortOptions = false, description = "Launches the JUnit Platform from the console.")
 class AvailableOptions {
 
 	private static final String CP_OPTION = "cp";
 
-	private final OptionParser parser = new OptionParser();
-
-	// General Purpose
-	private final OptionSpec<Void> help;
-	private final OptionSpec<Void> disableAnsiColors;
-	private final OptionSpec<Details> details;
-	private final OptionSpec<Theme> theme;
-	private final OptionSpec<Path> additionalClasspathEntries;
-	private final OptionSpec<Void> failIfNoTests;
-
-	// Reports
-	private final OptionSpec<Path> reportsDir;
-
-	// Selectors
-	private final OptionSpec<Path> selectedClasspathEntries;
-	private final OptionSpec<URI> selectedUris;
-	private final OptionSpec<String> selectedFiles;
-	private final OptionSpec<String> selectedDirectories;
-	private final OptionSpec<String> selectedPackages;
-	private final OptionSpec<String> selectedClasses;
-	private final OptionSpec<String> selectedMethods;
-	private final OptionSpec<String> selectedClasspathResources;
-
-	// Java Platform Module System Options
-	private final OptionSpec<Void> scanModulepath;
-	private final OptionSpec<String> selectedModules;
-
-	// Filters
-	private final OptionSpec<String> includeClassNamePattern;
-	private final OptionSpec<String> excludeClassNamePattern;
-	private final OptionSpec<String> includePackage;
-	private final OptionSpec<String> excludePackage;
-	private final OptionSpec<String> includeTag;
-	private final OptionSpec<String> excludeTag;
-	private final OptionSpec<String> includeEngine;
-	private final OptionSpec<String> excludeEngine;
-
-	// Configuration Parameters
-	private final OptionSpec<KeyValuePair> configurationParameters;
+	// --- General Purpose -------------------------------------------------
 
-	AvailableOptions() {
+	@Option(names = { "-h", "--help" }, usageHelp = true, description = "Display help information.")
+	private boolean helpRequested;
+
+	@Option(names = { "--h", "-help" }, help = true, hidden = true)
+	private boolean helpRequested2;
+
+	@Option(names = "--disable-ansi-colors", arity = "0..1", description = "Disable ANSI colors in output (not supported by all terminals).")
+	private boolean disableAnsiColors;
+
+	@Option(names = "-disable-ansi-colors", arity = "0..1", hidden = true)
+	private boolean disableAnsiColors2;
```

Comment:
If I remember correctly there were some unit tests that specified boolean options with an explicit value (e.g. `--disable-something=true`). Picocli assumes boolean options have zero arity (no args) unless we tell it it may have an optional argument by specifying `arity = "0..1"`.

Reply:
I can add a `paramLabel = "true|false"` that looks a bit better in the usage help.

Reply:
Hmm, `ConsoleDetailsTests`, right? Thanks for being so thorough but let's remove the boolean parameter from there. Otherwise, we'd have to do this for all boolean option including `--help` etc.

Reply:
The tests were thorough, not me! :-)

I’m fine with removing it but it means that input like `--disable-something=true` will be rejected with an `“unmatched parameter ‘true’”` error. 

This differs from the current behavior: the jopt-simple parser apparently accepts boolean options with and without parameters. 

Since there are tests that verify this, are you sure you want to modify this behavior?

Reply:
Normally I'd agree. But the test in question was actually an integration test that merely used the `ConsoleLauncher` to test something else. The tests for the parser did not verify this behavior and I wasn't even aware of it until now. I'll take the blame if people complain but IMHO this is really an edge case and I don't think it's likely that the change is going to break something.